### PR TITLE
Make cssnano compatible with browserify/webpack

### LINKED ISSUE #17: Weird warnings when consuming this with webpack
I have a specific need that not so specific: I am building a browser version of cssnext for the playground.
Not that cssnext embed cssnano I have weird notice when I make this build with webpack

It looks like some part of cssnano are being consumed via a require or something smiliar that make webpack think we want to consume some non js file.

```
WARNING in ./~/cssnano/index.js
Critical dependencies:
57:20-35 the request of a dependency is an expression
 @ ./~/cssnano/index.js 57:20-35

WARNING in ./~/cssnano/CHANGELOG.md
Module parse failed: /Users/MoOx/Sync/Development/cssnext/cssnext/node_modules/cssnano/CHANG
ELOG.md Line 1: Unexpected token ILLEGAL
You may need an appropriate loader to handle this file type.
| # 1.2.0
|
| * Better support for merging properties without the existance of a shorthand
 @ ./~/cssnano ^\.\/.*$

WARNING in ./~/cssnano/LICENSE-MIT
Module parse failed: /Users/MoOx/Sync/Development/cssnext/cssnext/node_modules/cssnano/LICEN
SE-MIT Line 1: Unexpected identifier
You may need an appropriate loader to handle this file type.
| Copyright (c) Ben Briggs <beneb.info@gmail.com> (http://beneb.info)
|
| Permission is hereby granted, free of charge, to any person
 @ ./~/cssnano ^\.\/.*$

WARNING in ./~/cssnano/bin/cmd.js
Module parse failed: /Users/MoOx/Sync/Development/cssnext/cssnext/node_modules/cssnano/bin/c
md.js Line 1: Unexpected token ILLEGAL
You may need an appropriate loader to handle this file type.
| #!/usr/bin/env node
|
| var fs = require('fs');
 @ ./~/cssnano ^\.\/.*$

WARNING in ./~/cssnano/README.md
Module parse failed: /Users/MoOx/Sync/Development/cssnext/cssnext/node_modules/cssnano/READM
E.md Line 1: Unexpected token ILLEGAL
You may need an appropriate loader to handle this file type.
| # cssnano [![Build Status](https://travis-ci.org/ben-eb/cssnano.svg?branch=master)][ci] [!
[NPM version](https://badge.fury.io/js/cssnano.svg)][npm] [![Dependency Status](https://gemn
asium.com/ben-eb/cssnano.svg)][deps] [![Gitter](https://badges.gitter.im/Join Chat.svg)](htt
ps://gitter.im/ben-eb/cssnano?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_co
ntent=badge)
|
| > A modular minifier, built on top of the [PostCSS] ecosystem.
 @ ./~/cssnano ^\.\/.*$

WARNING in ./~/cssnano/bin/usage.txt
Module parse failed: /Users/MoOx/Sync/Development/cssnext/cssnext/node_modules/cssnano/bin/u
sage.txt Line 1: Unexpected token {
You may need an appropriate loader to handle this file type.
| Usage: cssnano [input] [output] {OPTIONS}
|
| Options:
 @ ./~/cssnano ^\.\/.*$
```

From what I see, you are doing some dynamic require() which is not ideal for 2 things:
- browser build
- not future proof (see es6 imports)

But I get the lazy loading.
Maybe you should consider using a solution similar like cssnext does ? https://github.com/cssnext/cssnext/blob/0a5d06f4c1a71b5b62d450029d46dfa6cba0b333/index.js#L38-L93
I know this is verbose, but having hardcoded require() make this easier to consume, especially for webpack and browserify.

### PR DESCRIPTION
**Pull request by MoOx**
Close #17

### CODE REVIEW

**Comment by ben-eb** [comment]
Awesome. Thanks so much. :+1:

**Comment by MoOx** [comment]
Not that your issue was to try to use cssnano as a webpack loader. You do not need anything by default for .js :)

**Comment by ben-eb** [comment]
Ahh, yes I see. Still unfamiliar with webpack syntax. I'll release this as `1.3.0`.

**Comment by MoOx** [comment]
:+1:

File: tests/webpack.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    module: {
+        loaders: [{
+            test: /\.json$/,
+            loader: "json"
+        }]
+    },
+    // because client side doesn't have fs :)
+    node: {
+        fs: "empty"
+    }
```
Comment (sokra) [inline tests/webpack.js]: This is jsut a workaround for missbehaving packages.

You better should add
``` js
"browser": {
  "fs": false
}
```

to the package.json.

Comment (MoOx) [inline tests/webpack.js]: Ho, good to know !
Thanks for reporting !

**Pull request merged**